liferay-frontend/liferay-portal
# [LPS-116504, LPS-112471, LPS-112472] Updates clay:progressbar and clay:radio and adds props attribute to react:component

I know this has better chances at making it through separated and it would probably be the way to go, but I'm going to give this a go since most of the changes seem trivial...

**In this PR we:**
- Update `clay:progressbar` to use a java-based version instead of the old soy-based one. Checked 1 existing usage
- Update `clay:radio` to use a java-based version instead of the old soy-based one. Checked 1 existing usage
- Adds `props` attribute to `react:component` to better match expectations instead of calling it `data`
- Updates usages of `data` to `props`

**How to test it:**
- The update of `props` in `react:component` should be pretty obvious throughout CI... hoping to see issues there :fingers-crossed:
- Tested `clay:progressbar` in the User Associated Data widget. Check the [Exporting User Data](https://help.liferay.com/hc/en-us/articles/360029132051-Exporting-User-Data) tutorial (won't work with default user Test test, need to create another admin user for it).

<img width="1422" alt="Screen Shot 2020-07-20 at 16 18 06" src="https://user-images.githubusercontent.com/905006/87950072-1d692280-caa7-11ea-829e-fb32f87fffab.png">

- Tested `clay:radio` in the User Associated Data widget. Check the [Sanitizing User Data](https://help.liferay.com/hc/en-us/articles/360028819072-Sanitizing-User-Data#the-personal-data-erasure-screen) tutorial (won't work with default user Test test, need to create another admin user for it).
<img width="1237" alt="Screen Shot 2020-07-20 at 16 16 40" src="https://user-images.githubusercontent.com/905006/87950106-278b2100-caa7-11ea-9dc8-f5b05e2bf317.png">

**Not done:**
- Progressbars in the uad example are slightly misaligned... not sure if this was there before or is the result of this PR. Would need a followup
- Many  apis passing `props` to `react:component` are named `*DisplayContext.getSomethingData()`. I haven't renamed them here, but we might wanna for consistency... 🤔
